app/controllers/dashboard/protocols_controller.rb

```diff
@@ -177,31 +169,37 @@ def archive
   end
 
   def view_full_calendar
-    @service_request = @protocol.any_service_requests_to_display?
+    @service_request  = @protocol.any_service_requests_to_display?
+    arm_id            = params[:arm_id] if params[:arm_id]
+    page              = params[:page] if params[:page]
+
+    session[:service_calendar_pages]          = params[:pages] if params[:pages]
+    session[:service_calendar_pages][arm_id]  = page if page && arm_id
 
-    arm_id = params[:arm_id] if params[:arm_id]
-    page = params[:page] if params[:page]
-    session[:service_calendar_pages] = params[:pages] if params[:pages]
-    session[:service_calendar_pages][arm_id] = page if page && arm_id
-    @tab = 'calendar'
+    @tab    = 'calendar'
     @portal = params[:portal]
+
     if @service_request
       @pages = {}
       @protocol.arms.each do |arm|
-        new_page = (session[:service_calendar_pages].nil?) ? 1 : session[:service_calendar_pages][arm.id.to_s].to_i
-        @pages[arm.id] = @service_request.set_visit_page(new_page, arm)
+        new_page        = (session[:service_calendar_pages].nil?) ? 1 : session[:service_calendar_pages][arm.id.to_s].to_i
+        @pages[arm.id]  = @service_request.set_visit_page(new_page, arm)
       end
     end
+
     @merged = true
     respond_to do |format|
       format.js
     end
   end
 
   def display_requests
-    @protocol_role = @protocol.project_roles.find_by(identity_id: @user.id)
+    protocol_role           = @protocol.project_roles.find_by(identity_id: @user.id)
```

Comment:
We're going to render the partial and return it via JSON, but it has to be opened in the JS